src/Promitor.Agents.Scraper/Scheduling/ResourcesScrapingJob.cs

```diff
@@ -251,22 +251,63 @@ private void GetResourceScrapeDefinition(IAzureResourceDefinition resourceDefini
         }
 
         private async Task ScrapeMetrics(IEnumerable<ScrapeDefinition<IAzureResourceDefinition>> scrapeDefinitions, CancellationToken cancellationToken)
-        {
+        {   
             var tasks = new List<Task>();
+            var batchScrapingEnabled = this._azureMonitorIntegrationConfiguration.Value.MetricsBatching?.Enabled ?? false;
+            Logger.LogInformation("Parsed batch config: {Enabled}, {BatchSize}",  this._azureMonitorIntegrationConfiguration.Value.MetricsBatching.Enabled, this._azureMonitorIntegrationConfiguration.Value.MetricsBatching.MaxBatchSize);
```

Comment:
This can cause null reference

Reply:
Moved this down to only log when batching is enabled